### PR TITLE
Extract `:dependency-management` codecs from `:configuration-cache`

### PR DESCRIPTION
**Pull request by bamboo**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/core-configuration/configuration-cache/build.gradle.kts [hunk 1]
```diff
@@ -66,12 +62,18 @@ dependencies {
...
     implementation(projects.coreKotlinExtensions)
+    implementation(projects.dependencyManagementSerializationCodecs)
+    implementation(project(":enterprise-operations"))
     implementation(project(":execution"))
     implementation(project(":files"))
     implementation(project(":file-watching"))
+    implementation(project(":functional"))
     implementation(projects.flowServices)
     implementation(projects.guavaSerializationCodecs)
+    implementation(project(":hashing"))
```
Comment (alllex) [inline platforms/core-configuration/configuration-cache/build.gradle.kts:76]: 🎉

File: testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy [hunk 2]
```diff
@@ -53,7 +53,7 @@ abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
      * Change this whenever you add or remove subprojects for distribution-packaged plugins (lib/plugins).
      */
     int getPackagedPluginsJarCount() {
-        76
+        78
```
Comment (alllex) [inline testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:56]: 🤔  The subpprojects we extracted do not look like plugins. Should the core lib jar count (above) be incremented instead?

**Review by alllex — APPROVED** [review]
LGTM! With a small comment

File: testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy (hunk 2 above)
Comment (bamboo) [reply testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:56]: The produced jars do end up under `lib/plugins` 🤷

Comment (alllex) [reply testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:56]: Maybe we should take active steps towards removing the distinction in the tests, as recently [discussed](https://gradle.slack.com/archives/C03A0MT6S/p1715857946933779).

Comment (bamboo) [reply testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy:56]: Sure, however, this is the current structure this PR needs to comply with.

**Pull request merged**